flink-runtime/src/main/java/org/apache/flink/runtime/memory/MemoryManager.java

```diff
@@ -300,31 +255,21 @@ public void allocatePages(Object owner, List<MemorySegment> target, int numPages
 				allocatedSegments.put(owner, segmentsForOwner);
 			}
 
-			if (isPreAllocated) {
-				for (int i = numPages; i > 0; i--) {
-					MemorySegment segment = memoryPool.requestSegmentFromPool(owner);
-					target.add(segment);
-					segmentsForOwner.add(segment);
-				}
-			}
-			else {
-				for (int i = numPages; i > 0; i--) {
-					MemorySegment segment = memoryPool.allocateNewSegment(owner);
-					target.add(segment);
-					segmentsForOwner.add(segment);
-				}
-				numNonAllocatedPages -= numPages;
+			for (int i = numPages; i > 0; i--) {
+				MemorySegment segment = allocateManagedSegment(memoryType, owner);
+				target.add(segment);
+				segmentsForOwner.add(segment);
 			}
+			numNonAllocatedPages -= numPages;
 		}
 		// -------------------- END CRITICAL SECTION -------------------
 	}
 
 	/**
-	 * Tries to release the memory for the specified segment. If the segment has already been released or
-	 * is null, the request is simply ignored.
+	 * Tries to release the memory for the specified segment.
 	 *
-	 * <p>If the memory manager manages pre-allocated memory, the memory segment goes back to the memory pool.
-	 * Otherwise, the segment is only freed and made eligible for reclamation by the GC.
+	 * <p>If the segment has already been released or is null, the request is simply ignored.
```

Comment:
Remove completely?

I can change it to:
`If the segment has already been released, it is only freed. If it is null or has no owner, the request is simply ignored.`

Also, two more things came into my mind, we should not release budget if there is no such segment registered for this owner (e.g. double release) and we should always try to remove the segment even if it is freed because it might have been freed outside but not released.

Reply:
My bad, please ignore this comment.

I think I found some kind of typo with an extra 'if' in the doc/comment when I reviewed this PR locally in my IDE. But I cannot find it now. I probably have left this comment at wrong place when moving from IDE to GitHub.